src/containers/forms/PlanForm/helpers.ts

```diff
@@ -0,0 +1,339 @@
+      if (element.actionCode === 'BCC') {
+        thisAction = planActivities.BCC.action;
+        thisGoal = planActivities.BCC.goal;
+      }
+      if (element.actionCode === 'IRS') {
+        thisAction = planActivities.IRS.action;
+        thisGoal = planActivities.IRS.goal;
+      }
+      if (element.actionCode === 'Bednet Distribution') {
+        thisAction = planActivities.bednetDistribution.action;
+        thisGoal = planActivities.bednetDistribution.goal;
+      }
+      if (element.actionCode === 'Blood Screening') {
+        thisAction = planActivities.bloodScreening.action;
+        thisGoal = planActivities.bloodScreening.goal;
+      }
+      if (element.actionCode === 'Case Confirmation') {
+        thisAction = planActivities.caseConfirmation.action;
+        thisGoal = planActivities.caseConfirmation.goal;
+      }
+      if (element.actionCode === 'RACD Register Family') {
+        thisAction = planActivities.familyRegistration.action;
+        thisGoal = planActivities.familyRegistration.goal;
+      }
+      if (element.actionCode === 'Larval Dipping') {
+        thisAction = planActivities.larvalDipping.action;
+        thisGoal = planActivities.larvalDipping.goal;
+      }
+      if (element.actionCode === 'Mosquito Collection') {
+        thisAction = planActivities.mosquitoCollection.action;
+        thisGoal = planActivities.mosquitoCollection.goal;
+      }
+
+      // next put in values from the form
+      const actionFields: Partial<PlanAction> = {
+        description: element.actionDescription,
+        identifier: '',
+        prefix,
+        reason: element.actionReason as ActionReasonType,
+        timingPeriod: {
+          end: moment(element.timingPeriodEnd).format(DATE_FORMAT.toUpperCase()),
+          start: moment(element.timingPeriodStart).format(DATE_FORMAT.toUpperCase()),
+        },
+        title: element.actionTitle,
+      };
+
+      thisAction = Object.assign(thisAction, actionFields);
+
+      if (thisGoal.target[0]) {
+        const goalDetailQty: Partial<PlanGoaldetailQuantity> = {
+          value: element.goalValue,
+        };
+
+        const goalDetail: Partial<PlanGoalDetail> = {
+          detailQuantity: Object.assign(thisGoal.target[0].detail.detailQuantity, goalDetailQty),
+        };
+
+        const goalTarget: PlanGoalTarget = Object.assign(thisGoal.target[0], {
+          detail: goalDetail,
+          due: moment(element.goalDue).format(DATE_FORMAT.toUpperCase()),
+        });
+
+        const goalFields: Partial<PlanGoal> = {
+          description: element.goalDescription,
+          priority: element.goalPriority as GoalPriorityType,
+          target: [goalTarget],
+        };
+
+        thisGoal = Object.assign(thisGoal, goalFields);
+      }
+
+      actions.push(thisAction);
+      goals.push(thisGoal);
+    }
+  });
+
+  return {
+    action: actions,
+    goal: goals,
+  };
+}
+
+/**
+ * Get the plan name and title
+ * @param {any} event - the event object
```

Comment:
should be `@param {FormEvent} event - the event object`

Reply:
Yes it should. Good catch